pkg/operator/controllers/pullsecret/pullsecret_controller.go

```diff
@@ -192,10 +192,22 @@ func (r *Reconciler) ensureGlobalPullSecret(ctx context.Context, operatorSecret,
 		}
 
 		err = r.client.Create(ctx, secret)
+		if err != nil {
+			return secret, err
+		}
+
+		rc := pullsecret.NewRegistryClient()
+		err = rc.ValidatePullSecret(ctx, secret)
 		return secret, err
 	}
 
 	err = r.client.Update(ctx, secret)
+	if err != nil {
+		return secret, err
+	}
+
+	rc := pullsecret.NewRegistryClient()
+	err = rc.ValidatePullSecret(ctx, secret)
```

Comment:
Not exactly related to your change, but we might want to rename this whole function `rotateAndValidateGlobalPullSecret` or something similar since from your change it's clear that we're always going to rotate + validate it.

Reply:
Also, we should update operator statuses (progressing / degraded / etc) when we get an error validating.